src/adapters/node-http/monkeyPatch.ts

```diff
@@ -29,7 +38,103 @@ export const monkeyPatchVoidInputs = (appRouter: OpenApiRouter) => {
 
     const { inputParser } = getInputOutputParsers(mutation);
     if (instanceofZodTypeLikeVoid(inputParser)) {
-      (appRouter._def.mutations[mutationPath] as any).parseInputFn = zObject.parse.bind(zObject);
+      (mutation as any).parseInputFn = zObject.parseAsync.bind(zObject);
+    }
+  }
+};
+
+export const monkeyPatchParameterInputs = (appRouter: OpenApiRouter) => {
+  const { queries, mutations } = appRouter._def;
+
+  const zNumber = (schema: z.ZodType) =>
+    z.preprocess((arg) => {
+      if (typeof arg === 'string') return parseInt(arg);
+      return arg;
+    }, schema);
+  const zBoolean = (schema: z.ZodType) =>
+    z.preprocess((arg) => {
+      if (typeof arg === 'string') {
+        if (arg === 'true' || arg === '1') return true;
+        if (arg === 'false' || arg === '0') return false;
+      }
+      return arg;
+    }, schema);
+  const zDate = (schema: z.ZodType) =>
+    z.preprocess((arg) => {
+      if (typeof arg === 'string') return new Date(arg);
+      return arg;
+    }, schema);
+
+  for (const queryPath of Object.keys(queries)) {
+    const query = queries[queryPath]!;
+    const { openapi } = query.meta ?? {};
+    if (!openapi?.enabled) {
+      continue;
+    }
+
+    const { inputParser } = getInputOutputParsers(query);
+    if (instanceofZodTypeObject(inputParser)) {
+      const shape = inputParser.shape;
+      const shapeKeys = Object.keys(shape);
+      let zObject: z.AnyZodObject = inputParser;
+      shapeKeys.forEach((shapeKey) => {
+        const shapeSchema = shape[shapeKey]!;
+        if (instanceofZodTypeLikeNumber(shapeSchema)) {
+          zObject = zObject.merge(z.object({ [shapeKey]: zNumber(shapeSchema) }));
+          return;
+        }
+        if (instanceofZodTypeLikeBoolean(shapeSchema)) {
+          zObject = zObject.merge(z.object({ [shapeKey]: zBoolean(shapeSchema) }));
+          return;
+        }
+        if (instanceofZodTypeLikeDate(shapeSchema)) {
+          zObject = zObject.merge(z.object({ [shapeKey]: zDate(shapeSchema) }));
+          return;
+        }
+      });
```

Comment:
`throw` if there's an unexpected type which is unsupported?

Reply:
These inputs are already validated here: https://github.com/jlalmes/trpc-openapi/blob/9ed633cf596e42dc9c00501d383a6c68d14f1ab0/src/generator/schema.ts#L68